lib/src/ast/element.dart

```diff
@@ -45,11 +51,11 @@ abstract class ElementAst implements StandaloneTemplateAst {
   /// Create a new element AST from parsed source.
   factory ElementAst.parsed(
     SourceFile sourceFile,
-    NgToken beginToken,
+    NgToken openElementStart,
     NgToken nameToken,
-    int openTagEnd,
-    int closeTagStart,
-    NgToken endToken, {
+    NgToken openElementEnd, {
+    bool isVoidElement,
```

Comment:
do we need to track both things? That its void and its close element? Couldn't the former be a getter which returns `closeComplement == null`?